core/src/apps/nostr/sign_event.py

```diff
@@ -0,0 +1,54 @@
+from typing import TYPE_CHECKING
+
+from apps.common.keychain import auto_keychain
+
+if TYPE_CHECKING:
+    from trezor.messages import NostrEventSignature, NostrSignEvent
+
+    from apps.common.keychain import Keychain
+
+
+@auto_keychain(__name__)
+async def sign_event(msg: NostrSignEvent, keychain: Keychain) -> NostrEventSignature:
+    from ubinascii import hexlify
+
+    from trezor import TR
+    from trezor.crypto.curve import secp256k1
+    from trezor.crypto.hashlib import sha256
+    from trezor.messages import NostrEventSignature
+    from trezor.ui.layouts import confirm_value
+
+    from apps.common import paths
+
+    address_n = msg.address_n
+    created_at = msg.created_at
+    kind = msg.kind
+    tags = [[t.key] + ([t.value] if t.value else []) + t.extra for t in msg.tags]
+    content = msg.content
+
+    await paths.validate_path(keychain, address_n)
+
+    node = keychain.derive(address_n)
+    pk = node.public_key()[-32:]
+    sk = node.private_key()
+
+    title = TR.nostr__event_kind_template.format(kind)
+
+    # confirm_value on TR only accepts one single info item
+    # which is why we concatenate all of them here.
+    # This is not great, but it gets the job done for now.
+    tags_str = f"created_at: {created_at}"
+    for t in tags:
+        tags_str += f"\n\n{t[0]}: " + (f" {' '.join(t[1:])}" if len(t) > 1 else "")
+
+    await confirm_value(
+        title, content, "", "nostr_sign_event", info_items=[("", tags_str)]
+    )
+    serialized_tags = ",".join(
+        ["[" + ",".join(f'"{t}"' for t in tag) + "]" for tag in tags]
+    )
+    serialized_event = f'[0,"{hexlify(pk).decode()}",{created_at},{kind},[{serialized_tags}],"{content}"]'
```

Comment:
Can you comment on what is `0` here? Is that a part of the protocol or are we forcing some value?

Reply:
https://github.com/trezor/trezor-firmware/pull/4436/commits/126dc0b4a82bba4c131cc8f694a0eeece51d5f47#diff-1bfc3f5ba14a1d44de5030afbb9a530d7b9c4b307ced3789640e9234244379ddR47